src/library/scala/collection/immutable/OrderedMap.scala

```diff
@@ -0,0 +1,518 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala
+package collection
+package immutable
+
+/** This class implements an immutable map that preserves order using
+  * a hash map for the key to value mapping to provide efficient lookup,
+  * and a tree for the ordering of the keys to provide efficient
+  * insertion/modification order traversal and destructuring.
+  *
+  * By default insertion order (`OrderedMap.OrderBy.Insertion`)
+  * is used, but modification order (`OrderedMap.OrderBy.Modification`)
+  * can be used instead if so specified at creation.
+  *
+  * The `orderingBy(orderBy: OrderedMap.OrderBy): OrderedMap[K, V]` method
+  * can be used to switch to the specified ordering for the returned map.
+  *
+  * A key can be manually refreshed (i.e. placed at the end) via the
+  * `refresh(key: K): OrderedMap[K, V]` method (regardless of the ordering in
+  * use).
+
+  * Internally, an ordinal counter is increased for each insertion/modification
+  * and then the current ordinal is used as key in the tree map. After 2^32^
+  * insertions/modifications the entire map is copied (thus resetting the ordinal
+  * counter).
+  *
+  *
+  *  @tparam K the type of the keys contained in this map.
+  *  @tparam V the type of the values associated with the keys in this map.
+  *
+  * @author Odd Möller
+  * @version 2.13
+  * @since 2.13
+  * @define coll immutable ordered map
+  * @define Coll `immutable.OrderedMap`
+  */
+final class OrderedMap[K, +V] private (
+    private val ordering: OrderedMap.Ordering[K],
+    private val mapping: OrderedMap.Mapping[K, V],
+    private val ordinal: Int,
+    val orderedBy: OrderedMap.OrderBy)
+  extends AbstractMap[K, V]
+    with SeqMap[K, V]
+    with MapOps[K, V, OrderedMap, OrderedMap[K, V]]
+    with StrictOptimizedIterableOps[(K, V), Iterable, OrderedMap[K, V]]
+    with StrictOptimizedMapOps[K, V, OrderedMap, OrderedMap[K, V]] {
+
+  import OrderedMap._
+
+  override protected[this] def className: String = "OrderedMap"
+
+  override def mapFactory: MapFactory[OrderedMap] = OrderedMap
+
+  override val size = mapping.size
+
+  override def knownSize: Int = size
+
+  override def isEmpty = size == 0
+
+  def orderingBy(orderBy: OrderBy): OrderedMap[K, V] = {
+    if (orderBy == this.orderedBy) this
+    else new OrderedMap(ordering, mapping, ordinal, orderBy)
+  }
+
+  def updated[V1 >: V](key: K, value: V1): OrderedMap[K, V1] = {
+    mapping.get(key) match {
+      case e if ordinal == -1 && (orderedBy == OrderBy.Modification || e.isEmpty) ⇒
+        // Reinsert into fresh instance to restart ordinal counting, expensive but only done after 2^32 updates.
+        OrderedMap.empty[K, V](orderedBy) ++ this + (key → value)
```

Comment:
Please write out arrows as `->`. People can use ligature fonts if they want to view them in a cleaner shape; unicode arrows are hard for most people to type (not set up to do it).

Reply:
And it's on the way to be deleted ？

Reply:
indeed

Reply:
Fixed in commit 00b7f1e.